nixos/modules/tasks/network-interfaces.nix

```diff
@@ -541,6 +544,25 @@ in
       '';
     };
 
+    networking.machineId = mkOption {
+      default = null;
+      example = "b23d9f80c4494c00b632f2f6448aaa72";
+      type = types.nullOr types.str;
+      description = lib.mdDoc ''
+        The 128-bit machine ID of the machine, formatted as 32 hexadecimal characters.
+
+        You should try to make this ID unique among your machines.
+        The ID is generated randomly on boot, if it does not already exists.
```

Comment:
I'd add a hint akin to "You can generate a random id with `openssl rand -hex 16`."